src/main/java/yapion/serializing/ContextManager.java

```diff
@@ -90,7 +90,12 @@ YAPIONInfo is(Class<?> clazz) {
         if (is(clazz.getDeclaredAnnotation(YAPIONSaveExclude.class))) globalSave = false;
         globalSave = is(clazz.getDeclaredAnnotation(YAPIONSave.class));
 
-        yapionData = clazz.getDeclaredAnnotation(YAPIONData.class);
+        if (yapionData == null) {
+            yapionData = clazz.getDeclaredAnnotation(YAPIONData.class);
+        }
+        if (!isCascading()) {
+            yapionData = clazz.getDeclaredAnnotation(YAPIONData.class);
+        }
```

Comment:
I will look into making the cascading feature a more efficient one. in the serializer so that I do not need to create an Object each level until there is a cascading one. But that I will do myself.

Reply:
Great idea, I had not thought of that.

Reply:
Optimized with the commit: a87fe5bb4801b5c4259120cab61e26aed0c4211a